cpp/src/gandiva/function_registry.cc

```diff
@@ -35,415 +42,27 @@ using arrow::uint32;
-    NUMERIC_BOOL_DATE_VAR_LEN_TYPES(HASH32_SEED_SAFE_NULL_NEVER, hash32),
-    NUMERIC_BOOL_DATE_VAR_LEN_TYPES(HASH32_SEED_SAFE_NULL_NEVER, hash32AsDouble),
-
-    NUMERIC_BOOL_DATE_VAR_LEN_TYPES(HASH64_SAFE_NULL_NEVER, hash64),
-    NUMERIC_BOOL_DATE_VAR_LEN_TYPES(HASH64_SAFE_NULL_NEVER, hash64AsDouble),
-    NUMERIC_BOOL_DATE_VAR_LEN_TYPES(HASH64_SEED_SAFE_NULL_NEVER, hash64),
-    NUMERIC_BOOL_DATE_VAR_LEN_TYPES(HASH64_SEED_SAFE_NULL_NEVER, hash64AsDouble),
-
-    // utf8/binary operations
-    UNARY_SAFE_NULL_IF_NULL(octet_length, utf8, int32),
-    UNARY_SAFE_NULL_IF_NULL(octet_length, binary, int32),
-    UNARY_SAFE_NULL_IF_NULL(bit_length, utf8, int32),
-    UNARY_SAFE_NULL_IF_NULL(bit_length, binary, int32),
-    UNARY_UNSAFE_NULL_IF_NULL(char_length, utf8, int32),
-    UNARY_UNSAFE_NULL_IF_NULL(length, utf8, int32),
-    UNARY_UNSAFE_NULL_IF_NULL(lengthUtf8, binary, int32),
-
-    VAR_LEN_TYPES(BINARY_RELATIONAL_SAFE_NULL_IF_NULL, equal),
-    VAR_LEN_TYPES(BINARY_RELATIONAL_SAFE_NULL_IF_NULL, not_equal),
-    VAR_LEN_TYPES(BINARY_RELATIONAL_SAFE_NULL_IF_NULL, less_than),
-    VAR_LEN_TYPES(BINARY_RELATIONAL_SAFE_NULL_IF_NULL, less_than_or_equal_to),
-    VAR_LEN_TYPES(BINARY_RELATIONAL_SAFE_NULL_IF_NULL, greater_than),
-    VAR_LEN_TYPES(BINARY_RELATIONAL_SAFE_NULL_IF_NULL, greater_than_or_equal_to),
-
-    BINARY_RELATIONAL_SAFE_NULL_IF_NULL(starts_with, utf8),
-    BINARY_RELATIONAL_SAFE_NULL_IF_NULL(ends_with, utf8),
-
-    NativeFunction("upper", DataTypeVector{utf8()}, utf8(), kResultNullIfNull,
-                   "upper_utf8", NativeFunction::kNeedsContext),
-
-    NativeFunction("like", DataTypeVector{utf8(), utf8()}, boolean(), kResultNullIfNull,
-                   "gdv_fn_like_utf8_utf8", NativeFunction::kNeedsFunctionHolder),
-
-    NativeFunction("castDATE", DataTypeVector{utf8()}, date64(), kResultNullIfNull,
-                   "castDATE_utf8",
-                   NativeFunction::kNeedsContext | NativeFunction::kCanReturnErrors),
-
-    NativeFunction("to_date", DataTypeVector{utf8(), utf8(), int32()}, date64(),
-                   kResultNullInternal, "gdv_fn_to_date_utf8_utf8_int32",
-                   NativeFunction::kNeedsContext | NativeFunction::kNeedsFunctionHolder |
-                       NativeFunction::kCanReturnErrors),
-};  // namespace gandiva
-
 FunctionRegistry::iterator FunctionRegistry::begin() const {
-  return std::begin(pc_registry_);
+  return &(*pc_registry_.begin());
 }
 
 FunctionRegistry::iterator FunctionRegistry::end() const {
-  return std::end(pc_registry_);
+  return &(*pc_registry_.end());
 }
 
-FunctionRegistry::SignatureMap FunctionRegistry::pc_registry_map_ = InitPCMap();
+std::vector<NativeFunction> FunctionRegistry::pc_registry_;
 
-FunctionRegistry::SignatureMap FunctionRegistry::InitPCMap() {
+SignatureMap FunctionRegistry::pc_registry_map_ = InitPCMap();
+
+SignatureMap FunctionRegistry::InitPCMap() {
   SignatureMap map;
 
+#if 0
```

Comment:
if this is no longer required, please remove.

Reply:
removed the dead code.